### PR TITLE
Creation of json file to log changes and merge of rotate & mirror.

### LINKED ISSUE #108: Create json file to log changes.
#### Description:

Update the parametric methods so that a file is produced alongside the results detailing the changes made to each dwelling in each of the iterations in the study.

### LINKED ISSUE #114: Merge the mirror and rotate methods
#### Description:

Merge these two methods as its seems to be the same method repeated.

May be closed by #108

### PR DESCRIPTION
**Pull request by EKAdebo**
### Issues addressed by this PR

Closes #108
Closes #114 

The issues have been done. Please review. The code for the JSON QA file has been implemented and the files Rotate.cs and Mirror.cs have been merged to a file in the same folder Orientation.cs

### CODE REVIEW

**Comment by EKAdebo** [comment]
@BHoMBot check compliance

File: SAP_Engine/Modify/MergeJson.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            List<BH.oM.Environment.SAP.JSON.Opening> openings = new List<BH.oM.Environment.SAP.JSON.Opening>();
+            
+            foreach (var o in openings2)
+            {
+                if (names.Contains(o.Name))
+                {
+                    var test = openings1.Where(x=> x.Name == o.Name).ToList().First();
+                    
+                    if (test.Orientation == null) { test.Orientation = o.Orientation;}
+                    else { test.Orientation.Final = o.Orientation.Final; }
```
Comment (FraserGreenroyd) [inline SAP_Engine/Modify/MergeJson.cs]:
```suggestion
                    if (test.Orientation == null)
                        test.Orientation = o.Orientation;
                    else
                        test.Orientation.Final = o.Orientation.Final;
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Review by FraserGreenroyd — APPROVED** [review]
Some good work here @EKAdebo following today's workshops - we'll review in full in the next workshop so approving to unblock next steps.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core
@BHoMBot check serialisation

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am authorising dispensation to be granted on check ref. 15881654259

**Pull request merged**